tests/test_prolific.py

```diff
@@ -1,7 +1,17 @@
+"""Prolific module tests."""
+
+from inspect import isclass
 from unittest import mock
+from unittest.mock import MagicMock
```

Comment:
I can't make an argument that there's any doubt where `MagicMock` comes from, but in general importing just the module and then using `mock.Mock` and `mock.MagicMock` has been working well, so I'll make a pitch for that, especially since we've got a couple cases of that already.